core/controllers/collection_editor_test.py

```diff
@@ -170,11 +191,18 @@ def test_editable_collection_handler_put_can_access(self):
                 self.COLLECTION_ID))
         csrf_token = self.get_csrf_token_from_response(response)
 
-        json_response = self.put_json(
-            '%s/%s' % (
-                feconf.COLLECTION_EDITOR_DATA_URL_PREFIX,
-                self.COLLECTION_ID),
-            self.json_dict, csrf_token=csrf_token)
+        url = '%s/%s' % (feconf.COLLECTION_EDITOR_DATA_URL_PREFIX, self.COLLECTION_ID)
+        # Check PUT returns 500 when an exception is raised updating the
```

Comment:
I don't think raising 500 error codes is a good idea (Might be a problem with the underlying code). Appropriate 4xx error codes should be raised I think (/cc @seanlip)

Reply:
When I changed to 404, I get error as shown below which is called by the `put_json()` function in `test_utils.py`
![image](https://user-images.githubusercontent.com/28522090/54942110-5e136300-4f69-11e9-80b3-3fd91ef75e18.png)

Is this because I did not run the server while running the backend tests?

Reply:
Hmmm, if you scroll a bit higher, I think you'll see more logs corresponding to the 500 error. No you don't need a server running while running tests. The point I was trying to make is that server errors (500 error codes) aren't really desirable. We would like to fail gracefully with an appropriate 4xx error code. This might need tweaks in the original code.